transaction/views.py

```diff
@@ -38,3 +39,34 @@ def get_queryset(self):
 class TransactionExpenseAllView(TransactionMixin, ListView):
     def get_queryset(self):
         return super().get_queryset().filter(type=TransactionStaticData.EXPENSE)
+
+
+class RecurringTransactionMixin(CompanyFilterMixin):
+    lookup_field = 'id'
+    queryset = RecurringTransaction.objects.all()
+    serializer_class = RecurringTransactionSerializer
+
+
+class RecurringView(RecurringTransactionMixin, RetrieveCreateUpdateDestroyView):
+    def perform_destroy(self, instance):
+        instance.template.delete()
+        super().perform_destroy(instance)
+
+
+class RecurringAllView(RecurringTransactionMixin, ListView):
+    pass
+
+
+class RecurringActive(RecurringTransactionMixin, ListView):
+    def get_queryset(self):
+        return super().get_queryset().filter(end_date__gte=date.today())
```

Comment:
What if both the start and end date is in the future?

Reply:
Ops. Good point!